File: app/src/main/java/com/mancode/financetracker/database/viewmodel/CategoryViewModel.java
package com.mancode.financetracker.database.viewmodel;

import android.app.Application;

import com.mancode.financetracker.database.DataRepository;
import com.mancode.financetracker.database.entity.CategoryEntity;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

/**
 * Created by Manveru on 17.02.2018.
 */

public class CategoryViewModel extends AndroidViewModel {

    private DataRepository mRepository;
    private LiveData<List<CategoryEntity>> mAllCategories;


    public CategoryViewModel(@NonNull Application application) {
        super(application);
        mRepository = DataRepository.getInstance(application);
        mAllCategories = mRepository.getAllCategories();
    }

    public LiveData<List<CategoryEntity>> getAllCategories() {
        return mAllCategories;
    }

    public List<CategoryEntity> getIncomeCategories() {
        return mRepository.getIncomeCategories();
    }

    public List<CategoryEntity> getOutcomeCategories() {
        return mRepository.getOutcomeCategories();
    }
}

File: app/src/main/java/com/mancode/financetracker/viewmodel/CategoryViewModel.kt
package com.mancode.financetracker.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.mancode.financetracker.database.entity.CategoryEntity
import com.mancode.financetracker.utils.InjectorUtils

class CategoryViewModel(application: Application) : AndroidViewModel(application) {
    private val repository = InjectorUtils.getCategoriesRepository(application)

    val outcomeCategories: LiveData<List<CategoryEntity>> = repository.outcomeCategories
    val incomeCategories: LiveData<List<CategoryEntity>> = repository.incomeCategories

    fun insertCategory(name: String, categoryType: Int) {
        repository.insertCategory(
                CategoryEntity(0, name, categoryType)
        )
    }

    fun updateCategory(categoryEntity: CategoryEntity) {
        repository.updateCategory(categoryEntity)
    }

    fun deleteCategory(categoryEntity: CategoryEntity) {
        repository.deleteCategory(categoryEntity)
    }

    fun toggleHidden(categoryId: Int) {
        repository.toggleHidden(categoryId)
    }

    fun getCategoryDependencyCount(id: Int): LiveData<Int> = repository.getDependencyCount(id)
}
